File: backend/app/schemas/work.py
"""
作品相关Schema定义
"""
from pydantic import BaseModel, Field
from typing import Any, Dict, Optional


class WorkCreateRequest(BaseModel):
    """作品创建请求"""
    title: Optional[str] = Field(None, description="作品标题")
    templateId: str = Field(..., description="使用的模板ID")
    inputText: str = Field(..., description="用户输入的原始文本", min_length=1)
    infographicConfig: Dict[str, Any] = Field(..., description="完整的Infographic配置")
    userId: Optional[str] = Field(None, description="用户ID(可选)")


class WorkResponse(BaseModel):
    """作品响应"""
    id: int
    title: Optional[str]
    templateId: str
    inputText: str
    infographicConfig: Dict[str, Any]
    thumbnailUrl: Optional[str]
    createdAt: str
    updatedAt: str

File: frontend/src/api/work.ts
/**
 * 作品管理API
 */
import { apiClient } from './client'
import type { APIResponse } from './client'

/**
 * 作品创建请求
 */
export interface WorkCreateRequest {
  title: string
  templateId: string
  inputText: string
  infographicConfig: any
}

/**
 * 作品信息
 */
export interface Work {
  id: number
  title: string
  templateId: string
  inputText: string
  infographicConfig: any
  createdAt: string
  updatedAt: string
}

/**
 * 创建作品
 */
export async function createWork(data: WorkCreateRequest): Promise<APIResponse<Work>> {
  return apiClient.post('/works', data)
}

/**
 * 获取作品列表
 */
export async function getWorks(page = 1, pageSize = 20): Promise<APIResponse<{
  works: Work[]
  total: number
  page: number
  pageSize: number
}>> {
  return apiClient.get('/works', { params: { page, pageSize } })
}

/**
 * 获取单个作品
 */
export async function getWork(id: number): Promise<APIResponse<Work>> {
  return apiClient.get(`/works/${id}`)
}

/**
 * 删除作品
 */
export async function deleteWork(id: number): Promise<APIResponse<null>> {
  return apiClient.delete(`/works/${id}`)
}
